src/Liquid/AbstractBlock.php

```diff
@@ -112,24 +112,16 @@ public function parse(array &$tokens)
 	 *
 	 * @param string $token
 	 */
-	protected function whitespaceHandler($token)
+	protected function whitespaceHandler($tokenParts)
 	{
-		/*
-		 * This assumes that TAG_START is always '{%', and a whitespace control indicator
-		 * is exactly one character long, on a third position.
-		 */
-		if (mb_substr($token, 2, 1) === Liquid::get('WHITESPACE_CONTROL')) {
+		if ($tokenParts[2] === Liquid::$config['WHITESPACE_CONTROL'] || $tokenParts[5] === Liquid::$config['WHITESPACE_CONTROL']) {
```

Comment:
A superior option would be to fetch it once in the constructor and reuse later.
```
$this->whitespace_control = Liquid::get('WHITESPACE_CONTROL');
```

Reply:
The new regex outputs dedicated array indices for these,  so reverting this would be meaningless.